### PR TITLE
Improve handling of mismatching /BaseFont and /FontName entries for non-embedded fonts (issue 7454)

### LINKED ISSUE #7454: wingdings font not rendered properly
Link to PDF file (sorry for not being able to attach file link since it's confidential pdf content):

Configuration:
- Web browser and its version: chrome (50.0.2661.94 m) , IE (11) , Firefox (46)
- Operating system and its version: Windows 7
- PDF.js version:1.5.309
- Is an extension:no

What is the expected behavior? (add screenshot)
checkboxok

What went wrong? (add screenshot)
checkboxko

The checkboxes are rendered fine with adobeReader and with chrome pdfjs Extension but cant renderer wingdings fonts properly .

### PR DESCRIPTION
**Pull request by Snuffleupagus**
This patch is the result of me going through some old issues regarding non-embedded Wingdings support.

There's a few different things wrong in the referenced PDF document:
 - The /BaseFont and /FontName entries don't agree on the name of the fonts, with one font using `/BaseFont /Wingdings-Regular` and `/FontName /wg09np` which obviously makes no sense.
   To address this we'll compare the font-names against our lists of known ones and ignore /FontName entries that don't make sense iff the /BaseFont entry is a known font-name.
 - The non-embedded Wingdings font also set an incorrect /Encoding, in this case /MacRomanEncoding, which should have been fixed by PR #16465. However this doesn't work since the font has *bogus* font-flags, that fail to categorize the font as Symbolic.
   To address this we'll also compare the font-name against the list of known symbol fonts.

### CODE REVIEW

File: src/core/evaluator.js [hunk 1]
```diff
@@ -4257,10 +4255,21 @@ class PartialEvaluator {
...
-          fontName = baseFont;
+        if (fontNameStr && baseFontStr) {
+          // Workaround for cases where e.g. fontNameStr = 'Arial' and
+          // baseFontStr = 'Arial,Bold' (needed when no font file is embedded).
+          if (baseFontStr.startsWith(fontNameStr)) {
+            fontName = null;
+          }
+          // Workaround for cases where e.g. fontNameStr = 'wg09np' and
+          // baseFontStr = 'Wingdings-Regular' (fixes issue7454.pdf).
+          if (
```
Comment (calixteman) [inline src/core/evaluator.js:4266]: You could replace the `if` by a `else if`.

**Review by calixteman — APPROVED** [review]
LGTM. Thank you.

File: src/core/evaluator.js (hunk 1 above)
Comment (calixteman) [reply src/core/evaluator.js:4264]: You could merge the new if if the above one.

Comment (Snuffleupagus) [reply src/core/evaluator.js]: Sure, I was just a little bit worried about overall readability by doing that.

Comment (calixteman) [reply src/core/evaluator.js]: The meaning of the first two checks is pretty obvious and are here to allow the following ones, so from my pov it's ok.

**Pull request merged**